examples/remote/binary_dependencies/cargo/crates.bzl

```diff
@@ -9,6 +9,144 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # bui
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")  # buildifier: disable=load
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")  # buildifier: disable=load
 
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dependencies for the Rust targets of that package.
+_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+        "ferris-says": "@remote_binary_dependencies__ferris_says__0_2_0//:ferris_says",
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dependencies for the Rust targets of that package.
+_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dev dependencies for the Rust targets of that package.
+_DEV_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dev dependencies for the Rust targets of that package.
+_DEV_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+def crates(deps):
```

Comment:
Should this be called "dependencies"? "crates" communicates almost nothing about the semantics.

Reply:
I kinda liked the vernacular of `crates` and I think contextually it's clear that the crates are dependencies. I can change it if you feel strongly but I would ask for more thoughts on this first.

Reply:
The crates aren't even actually dependencies. They're build targets. What this is doing is resolving a list of crate names to the build targets. 

What do you think about "crate_targets"

Reply:
I think `crate_targets` is implied by `crates`. The macro is returning a list of targets already. It feels unnecessary to make the name more verbose. Is there another short one word name we can use?

Reply:
I prefer verbosity especially when we're dealing with magical stuff, so I'm going to insist upon a verbose name. I suggest below the name "resolve_crate_dependences" and "resolve_crate_dependences_here" for the non-magic and the magic version

Reply:
Discussion on this is split into multiple comment sections, but I think we can both live with "crate_deps".

Reply:
So `crate_deps` confirmed. Leaving now... `package_crate_deps`?

Reply:
That sounds fine to me

Reply:
No, `all_crate_deps` I think is more correct since `package_name` is a parameter now.

Reply:
Ah dang, we both replied at the same time 😅

Your call. I'll make the change for `all_crate_deps` but can still change.

Reply:
Updated the names

Reply:
iiuc, "all_crate_deps" isn't right for either the parameterized or unparameterized version. package_name is used to specify which package's dependencies are looked up, whereas the unparameterized version assumes that the current build target is a package in the resolved dependencies.

is that understanding incorrect?

Reply:
I don't think that's quite right, they both assume the current package unless the `package_name` is set to something otherwise. `crate_deps` is for looking up individual dependencies where `all_crate_deps` is for returning all the dependencies associated with `package_name` (again, defaulting to the current package).

Reply:
Oh, my bad, I've gotten confused between the discussions.

`all_crate_deps` is fine.

Reply:
Marking resolved.